src/Symfony/Component/DependencyInjection/Compiler/RandomizePrivateServiceIdentifiers.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Alias;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Randomizes private service identifiers, effectively making them unaccessable from outside.
+ *
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class RandomizePrivateServiceIdentifiers implements CompilerPassInterface
```

Comment:
Not sure how DX is involved here. So maybe but I don't what it'd mean :)

> Forget about randomizing, and compile different method names

maybe :) the goal is to have private services be hidden from the outside, no interference with the public interface.
How it's done is an implementation detail We should chose the simplest one...  :)